### ISSUE TITLE
Will super-tabs support for loading a new set of tabs?

### ISSUE THREAD

**Issue by AlGantori** [zyra/ionic-super-tabs#247]
I have not tested "super-tabs" but I don' t see anything the following tab specification, much like standard ionic tabs, a way to tell that page2 could be another set of tabs and that perhaps to coder wants page2 to be "pushed" rather than just a tab with a page activated.

```
<super-tabs>
  <super-tab [root]="page1" title="First page" icon="home"></super-tab>
  <super-tab [root]="page2" title="Second page" icon="pin"></super-tab>
  <super-tab [root]="page3" title="Third page" icon="heart"></super-tab>
</super-tabs>
```

In standard ionic when I attempt to have page2.html be another 
```
<tabs>
...
</tabs> 
```
specification, I do get the default page of page2.html activated/displayed but the tabs and icons remain the same.

If "super-tabs" supports loading a different set of tabs please let me know and if there is sample code and how it should be done, please share.

BTW, I have also tried having page2 be a temporary ordinary page which then via setTimeout 10ms, attempts to setRoot( TabsSet2 ) but without success, the first tabs defined in TabsSet1 remain in view.

To summarize, the desired effect would be, for example, select or swipe to the second tab and a new set of tabs/icons appears.  The coder should have the ability to specify statically or control programmatically if they are being pushed and thus a "back arrow" would appear on the top left in the navigation bar (much like pushing a page).

**Comment by eduardoRoth** [comment]
Hi @AlGantori , here's an approach you can try:

**tabsParent.html** *<-- The parent tabs container*
```
<super-tabs id="parentTabsId" #parentTab toolbarBackground="light" tabsPlacement="bottom">
        <super-tab id="tab-1" [root]="tab1Root" [title]="tab1Title" icon="home"></super-tab>
        <super-tab id="tab-2" [root]="tab2Root" [title]="tab2Title" icon="person"></super-tab>
        <super-tab id="tab-3" [root]="tab3Root" [title]="tab3Title" icon="cog"></super-tab>
</super-tabs>
```
*tab2Root* is the component for out child tabs container

**tabsChild.html** *<-- This is going to be the child tabs container*
```
<ion-header>
    <ion-toolbar>
        <ion-buttons end>
            <button ion-button clear color="light" (tap)="goBack()">Go Back</button>
        </ion-buttons>
    </ion-toolbar>
</ion-header>

<ion-content>
    <super-tabs id="childTabsId" #childTab toolbarBackground="light" tabsPlacement="bottom">
        <super-tab id="tab-child-1" [root="tabChild1Root" [title]="tabChild1Title" icon="phone"></super-tab>
        <super-tab id="tab-child-2" [root]="tabChild2Root" [title]="tabChild2Title" icon="contacts">
    </super-tab>
</super-tabs>
```

**tabsChild.component.ts**
```
    @ViewChild('childTab') childTab: SuperTabs;
    constructor(private superTabs : SuperTabsController){}

    ionViewWillEnter() {
        this.childTab.showToolbar(false);
    }

    ionViewDidEnter() {
        this.superTabs.showToolbar(false, 'parentTabsId');
        this.childTab.showToolbar(true);
    }

    goBack() {
        this.childTab.showToolbar(false);
        this.superTabs.showToolbar(true, 'parentTabsId');
        this.superTabs.slideTo('tab-1', 'parentTabsId');
    }
```

It's not a perfect approach, it flickers a bit when you enter or exit the page because of hiding and showing the tabs containers (maybe you could use an animation handler to kind of fix this), and you have to click a button or text to go back to parent tabs (although you could create a gesture handler to go back).

Hope this helps!

**Comment by ihadeed** [comment]
Dynamically adding/removing tabs after it's been rendered is something that I intend to implement, but I don't have a date estimate yet.

**Comment by AlGantori** [comment]
Wanted to share the following: it occurred to me that nested tabs have a conflict with user interface experience and flow and perhaps even more so with swipeable tabs like super-tabs. 
Even when implemented it produces a non-intuitive flow.
For example, what happens when one travels to a "child" tab set? will swiping apply to the child and how will the user swipe out to the parent tabs :) ?

A bit different example, consider the following app snapshot, I run into, which combines tabs at the bottom and swipeable pages indicator at the top
Swipeable tabs in this situation would create a confusion as to what user is swiping.

![tabs and swipeable pages](https://user-images.githubusercontent.com/11699920/35547148-3fb272f8-052c-11e8-9d8a-02d096e61122.png)

**Comment by nelutzu91** [comment]
> Dynamically adding/removing tabs after it's been rendered is something that I intend to implement, but I don't have a date estimate yet.

Hello @ihadeed ,
did you find the time for this feature? 
Or is still ongoing?

**Comment by ihadeed** [comment]
This will be part of the upcoming version that works with Ionic 4; there is a possibility that it might work on Ionic 3 as well.

**Comment by nelutzu91** [comment]
> This will be part of the upcoming version that works with Ionic 4; there is a possibility that it might work on Ionic 3 as well.

Could you give the an example on how to do it in Ionic 3?
Thanks.
